src/main/java/net/consensys/orion/http/handler/send/SendRequest.java

```diff
@@ -80,6 +83,10 @@ public SendRequest(final byte[] rawPayload, final String from, final String[] to
     this.rawPayload = rawPayload;
     this.from = from;
     this.to = to;
+
+    if (from != null && (to == null || to.length == 0) && privacyGroupId().isEmpty()) {
```

Comment:
I can't quite get my head around this logic - shouldn't it say:
* From must be populated AND either TO or PrivacyGroupId must be populated?
(Isn't privacyGroupId() guaranteed to be empty at this point?

Reply:
You are right. At this point `privacyGroupId().isEmpty()` is always true :)